server/admin_management/api/app/routers/router_guards.py

```diff
@@ -234,10 +213,8 @@ async def enforce_self_grant_guard(
     """
     LOGGER.debug(f"enforce_self_grant_guard: requester - {requester}")
     LOGGER.debug(f"enforce_self_grant_guard: target_user - {target_user}")
-    is_same_user_name = requester.user_name == target_user.user_name
-    is_same_user_type_code = requester.user_type_code == target_user.user_type_code
 
-    if is_same_user_name and is_same_user_type_code:
+    if requester.user_guid == target_user.user_guid:
```

Comment:
updated to check user_guid, because we can expect both requester and target_user will have the user_guid